gradle-plugins/compose-preview-runtime/src/main/kotlin/org/jetbrains/compose/desktop/preview/runtime/ComposePreviewRunner.kt

```diff
@@ -0,0 +1,35 @@
+package org.jetbrains.compose.desktop.preview.runtime
+
+import kotlin.reflect.KProperty1
+
+class ComposePreviewRunner {
+    companion object {
+        private const val PREVIEW_ANNOTATION_FQ_NAME = "androidx.compose.ui.tooling.desktop.preview.Preview"
+
+        @JvmStatic
+        fun main(args: Array<String>) {
+            val classLoader = ComposePreviewRunner::class.java.classLoader
+
+            val previewFqName = args[0]
+            val previewClassFqName = previewFqName.substringBeforeLast(".")
+            val previewMethodName = previewFqName.substringAfterLast(".")
+            val previewClass = classLoader.loadClass(previewClassFqName)
+            val previewMethod = previewClass.methods.find { it.name == previewMethodName }
+                ?: error("Could not find method '$previewMethodName' in class '${previewClass.canonicalName}'")
+
+            val content = previewMethod.invoke(previewClass)
+            val previewAnnotation = previewMethod.annotations.find { it.annotationClass.qualifiedName == PREVIEW_ANNOTATION_FQ_NAME }
+                ?: error("Could not find '$PREVIEW_ANNOTATION_FQ_NAME' annotation on '$previewClassFqName#$previewMethodName'")
+            val environmentKClassProperty = previewAnnotation.annotationClass.members.find { it is KProperty1<*, *> && it.name == "environment" }
+                as KProperty1<Any, Class<*>>
+            val environmentClass = environmentKClassProperty.get(previewAnnotation)
+            val previewEnvironment = environmentClass
+                .getDeclaredConstructor()
+                .newInstance()
+            val showMethod = previewEnvironment.javaClass
```

Comment:
We rely on JVM reflection because default Kotlin reflection doesn't have required functionality?

Reply:
Kotlin reflection probably has required functionality, but the annotation property `environment: KClass<ComposePreviewEnvironment>` becomes `Class<ComposePreviewEnvironment>` in bytecode.
Converting it back to `KClass` is possible, but the code becomes less straightforward anyways IMO